### ISSUE TITLE
Projects don't load anymore (OpenRefine 3.7.6)

### ISSUE THREAD

**Issue by trnstlntk** [OpenRefine/CommonsExtension#96]
I hadn't used the Commons extension for a long while (probably since a few OpenRefine versions ago). I now tried to load projects based on Commons categories, and then projects refuse to load correctly.
I'm on MacOS using OpenRefine 3.7.6 and the Commons extension 0.1.1.

Screenshot after the preview screen:
![image](https://github.com/OpenRefine/CommonsExtension/assets/9881645/1332462d-23bc-46b8-8937-7d95b27b6cb8)

Interestingly, it does show that there are 48 rows and the project can also be exported (I tried to an excel sheet, and that works correctly, I get a sheet with file names then).

**Comment by trnstlntk** [comment]
Note: I tried loading many different Commons categories, and with different parameters in the preview screen. No difference, all the same result. The preview screen looks good.

**Comment by wetneb** [comment]
This looks like an error linked to the thumbnail rendering.
Can you share the category you are trying to import and look in your browser's console if to see if you see errors there? (Right click on anywhere in the page, select "Inspect" and then go to the "Console" tab)

**Comment by trnstlntk** [comment]
Thanks, that's a fast response :-)

Console says this:
```
JQMIGRATE: Migrate is installed with logging active, version 3.4.0
project-bundle.js:59615 Uncaught TypeError: Cannot read properties of undefined (reading 'identifierSpace')
    at ThumbnailReconRenderer.render (project-bundle.js:59615:71)
    at DataTableCellUI._render (project-bundle.js:46509:36)
    at new DataTableCellUI (project-bundle.js:46466:8)
    at renderRow (project-bundle.js:45731:9)
    at DataTableView._renderDataTables (project-bundle.js:45743:5)
    at DataTableView.render (project-bundle.js:45405:8)
    at project-bundle.js:45750:10
    at Object.success (project-bundle.js:38198:9)
    at fire (project-bundle.js:3502:31)
    at Object.fireWith [as resolveWith] (project-bundle.js:3632:7)
```
I was loading the categories [Rotterdam in the 17th Century](https://commons.wikimedia.org/wiki/Category:Rotterdam_in_the_17th_century) and [Rotterdam in the 18th Century](https://commons.wikimedia.org/wiki/Category:Rotterdam_in_the_18th_century).

**Comment by trnstlntk** [comment]
If it helps (maybe unrelated, maybe not): I have now loaded a project with Commons files from clipboard, and reconciled them. I noticed two things (I can create separate issues if these are unrelated):

* Commons reconciliation service was not in the preset list of reconciliation services, while I think I recall it would come with the Commons extension (I uninstalled and reinstalled OpenRefine and the extension before testing again, so I had a fresh system)
* No thumbnails are generated in the grid after reconciling, even though the files are simple jpegs which were/are supported

**Comment by trnstlntk** [comment]
On PAWS, which is also upgraded to 3.7.6, it works just fine 🤯

<img width="790" alt="image" src="https://github.com/OpenRefine/CommonsExtension/assets/9881645/634e4327-2dd7-44cb-a507-f67945adaaf1">

**Comment by wetneb** [comment]
Is the Commons reconciliation service correctly registered in your OpenRefine? Looking at the error message I'd say it's likely that it's not there. You could try removing it and adding it again, perhaps.

**Comment by trnstlntk** [comment]
That's right, I removed it during a training. 🤦‍♀️

I re-added it - project now loads with the file names reconciled, at least, but no thumbnails yet.
![image](https://github.com/OpenRefine/CommonsExtension/assets/9881645/4037ee66-5d53-4334-9451-7768f036b966)

**Comment by trnstlntk** [comment]
I have burned a few candles, I spoke in tongues, prayed to seventeen gods, and lo and behold, this issue is now solved for me.
<img width="1334" alt="image" src="https://github.com/OpenRefine/CommonsExtension/assets/9881645/2c0aceb4-245a-41d5-ba1e-64db9c5b8600">


I have absolutely no idea where it went wrong. The last things I did, just if anyone ever discovers this issue because they have the same thing going on:
* I did fresh installs of OpenRefine 3.7.4 and then 3.7.6 (removing all my old projects and settings, reinstalling the Commons extension). That did not help.
* I am on Mac OS, I tried downloading and starting the Linux version - the above issue with disappeared thumbnails did not go away.
* I then re-installed the 3.7.6 MacOS version and placed all my old projects, extensions and settings back. 
* Next, I opened an old Wikimedia Commons focused project. Its grid looked broken (only part of one row was displayed)
* I removed and then readded the Wikimedia Commons reconciliation service
* And then it worked 😳

**Comment by wetneb** [comment]
In any case it looks like the code is missing some safeguards, it really should not fail while rendering a cell in a way that aborts the rendering of the rest of the grid, so I'd say it would be worth keeping this issue open for that.

**Comment by lirazelf** [comment]
In addition to the above, re-adding the reconciliation service manually partially fixes the problem, and the project loads, but doesn't show thumbnails (except when I mouse over the reconciled files.)

**Comment by LNBelo** [comment]
> I hadn't used the Commons extension for a long while (probably since a few OpenRefine versions ago). I now tried to load projects based on Commons categories, and then projects refuse to load correctly. I'm on MacOS using OpenRefine 3.7.6 and the Commons extension 0.1.1.
> 
> Screenshot after the preview screen: ![image](https://private-user-images.githubusercontent.com/9881645/277982314-1332462d-23bc-46b8-8937-7d95b27b6cb8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMwNjU3MzMsIm5iZiI6MTcxMzA2NTQzMywicGF0aCI6Ii85ODgxNjQ1LzI3Nzk4MjMxNC0xMzMyNDYyZC0yM2JjLTQ2YjgtODkzNy03ZDk1YjI3YjZjYjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTRUMDMzMDMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjA0NmRiYzRhZjgzZjAxYTlmMzg5OGE5ZTUyNzFmNDAyMDU4NTVmMGY0NmJhNmI2MzE0ODlhZjZkNmE4YjhjNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.hOPCqsGy7OT6UqyM3RmrozqdglT9Ian4UZrchtx-vs0)
> 
> Interestingly, it does show that there are 48 rows and the project can also be exported (I tried to an excel sheet, and that works correctly, I get a sheet with file names then).

I just created a new bot account on Commons and opened paws to insert SDC from wiktext. When using the Wikimedia Commons extension for a category, exactly the same screen with one line and one column appeared. I noticed that only the Wikidata reconciliation service was installed, after installing the Wikimedia Commons reconciliation service everything started working normally again.

**Comment by lokal-profil** [comment]
Paws uses 3.7.9, whereas the patch for this is only available from [3.8.0-beta1](https://github.com/OpenRefine/OpenRefine/releases/tag/3.8-beta1)

**Comment by wetneb** [comment]
This should be solved with OpenRefine 3.8 and the Commons extension 0.1.2. Some thumbnails will likely still not appear, but it will not break the displaying of the entire grid. Perhaps we can leave this issue open to fix the underlying problem (of some thumbnails not displaying).
